DM-50064: Roundtable GKE version update and Fuse Driver

dspeck1 opened the pull request:
Updates GKE module to more recent version.  Adds support for enabling FUSE driver in GKE and enables in dev.

jonathansick commented:
```
Error: Cycle: module.gke.module.gke.google_container_node_pool.pools["zookeeper-pool"], module.gke.module.gke.google_container_node_pool.pools["zookeeper-pool"] (destroy deposed f93ce77c), module.gke.module.gke.local.cluster_endpoint_for_nodes (expand), module.gke.module.gke.google_compute_firewall.intra_egress (expand), module.gke.module.gke.google_container_node_pool.pools (expand), module.gke.module.gke.google_container_node_pool.pools["core-pool"], module.gke.module.gke.google_container_node_pool.pools["core-pool"] (destroy deposed 9f3f425f), module.gke.module.gke.google_container_cluster.primary, module.gke.module.gke.google_container_node_pool.pools["kafka-pool"] (destroy deposed ad221f8d), module.gke.module.gke.random_shuffle.available_zones[0] (destroy), module.gke.module.gke.google_container_node_pool.pools["kafka-pool"]

```

Does this mean terraform is erroring on deleting the zookeeper pool? I think it's ok to destroy the zookeeper pool because Sasquatch no longer uses zookeeper and we're not using it for node taints.

roceb commented:
At least we got to clean up some stuff from this.

dspeck1 commented:
Fixed the cycle error by removing `available_zones` from state because there was a circular dependency.  Updated state with the new elements so that a replacement is not needed now.  @jonathansick - I will message you in the morning then apply.
